libfwupdplugin/fu-hidraw-device.c

```diff
@@ -353,6 +353,115 @@ fu_hidraw_device_get_feature(FuHidrawDevice *self,
 #endif
 }
 
+/**
+ * fu_hidraw_device_set_report:
+ * @self: a #FuHidrawDevice
+ * @buf: (not nullable): a buffer to use, which *must* be large enough for the request
+ * @bufsz: the size of @buf
+ * @flags: some #FuIOChannelFlags, e.g. %FU_IO_CHANNEL_OPEN_FLAG_NONBLOCK
+ * @error: (nullable): optional return location for an error
+ *
+ * Do a HID SetOutputReport request.
+ *
+ * Returns: %TRUE for success
+ *
+ * Since: 2.0.14
+ **/
+gboolean
+fu_hidraw_device_set_report(FuHidrawDevice *self,
+			    const guint8 *buf,
+			    gsize bufsz,
+			    FuIOChannelFlags flags,
+			    GError **error)
+{
+#ifdef HAVE_HIDRAW_H
+	g_autofree guint8 *buf_mut = NULL;
+#endif
+
+	g_return_val_if_fail(FU_IS_HIDRAW_DEVICE(self), FALSE);
+	g_return_val_if_fail(buf != NULL, FALSE);
+	g_return_val_if_fail(error == NULL || *error == NULL, FALSE);
+
+#ifdef HAVE_HIDRAW_H
+	fu_dump_raw(G_LOG_DOMAIN, "SetReport", buf, bufsz);
+	buf_mut = fu_memdup_safe(buf, bufsz, error);
+	if (buf_mut == NULL)
+		return FALSE;
```

Comment:
I don't understand why we have to build a mutable buffer here -- fu_udev_device_write() takes a `const guint8 *buf`

Reply:
ah! you're right, I just copied it from `fu_hidraw_device_set_feature` but here we don't need it.